src/main/scala/zio/prelude/package.scala

```diff
@@ -112,4 +114,21 @@ package object prelude
     def lessOrEqual(that: A)(implicit ord: Ord[A]): TestResult =
       assert(self)(isLessThanEqualTo(that))
   }
+
+  implicit class AnySyntax[A](private val a: A) extends AnyVal {
+
+    @silent("side-effecting nullary methods are discouraged")
+    /* Ignores the value, if you explicitly want to do so and avoids "Unused value" compiler warnings. */
+    def ignore: Unit = ()
+
+    /** Applies function `f` to a value `a`, like `f(a)`, but in flipped order and doesn't need parentheses. Can be chained, like `x |> f |> g`. */
+    def |>[B](f: A => B): B = f(a)
+
+    /** Applies the function `f` to the value `a`, ignores the result, and returns the original value `a`. Practical for debugging, like `x.someMethod.tee(println(_)).someOtherMethod...` . Similar to the `tee` UNIX command. */
+    def tee[B](f: A => B): A = {
```

Comment:
I think this would be called `tap`. Given that this function is only meaningful if `f` is side effecting not sure it is something we want to be adding to every method in a functional programming library. Also could eliminate the `B` type parameter and use `Any` for the result type.

Reply:
> `tap`

OK, no problem

> `B` -> `Any`

Good point!

> only meaningful if `f` is side effecting

Yeah, it's a pickle. It seemed like a practical thing, especially for debugging purposes. But you're right, ZIO prelude is supposed to be about pure functional programming, so I definitely wouldn't be insisting on including it.

So what do you say? `tap` goes out?